src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -0,0 +1,180 @@
+            }
+            catch
+            {
+                _logger.Error("Broker Operation Failed to complete.");
+                throw;
+            }
+#pragma warning restore 4014
+            await readyForResponse.WaitAsync().ConfigureAwait(false);
+            return resultEx;
+        }
+
+        private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerPayload)
+        {
+            _brokerHelper.SayHelloToBroker(_activity);
+
+            Context mContext = Application.Context;
+
+            brokerPayload[BrokerParameter.BrokerAccountName] = AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            _logger.InfoPii(
+                "It switched to broker for context: " + mContext.PackageName + " login hint: " +
+                AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.BrokerAccountName),
+                "It switched to broker for context");
+
+            // Don't send silent background request if account information is not provided
+            bool hasAccountNameOrUserId = !string.IsNullOrEmpty(AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.BrokerAccountName)) ||
+                !string.IsNullOrEmpty(AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username));
+            if (hasAccountNameOrUserId)
+            {
+                _logger.Verbose("User is specified for silent token request");
+                string silentResult = _brokerHelper.GetAuthTokenSilently(brokerPayload, _activity);
+                resultEx = CreateMsalTokenResponseFromResult(silentResult);
+                readyForResponse.Release();
+                return;
+            }
+            else
+            {
+                _logger.Verbose("User is not specified for silent token request");
+            }
+
+            _logger.Verbose("Starting Authentication Activity");
+
+            if (resultEx == null)
+            {
+                _logger.Verbose("Initial request to broker");
+                // Log the initial request but not force a prompt
+            }
+
+            // onActivityResult will receive the response for this activity.
+            // Lauching this activity will switch to the broker app.
+            Intent brokerIntent = _brokerHelper.GetIntentForInteractiveBrokerRequest(brokerPayload, _activity);
+            if (brokerIntent != null)
+            {
+                try
+                {
+                    _logger.Verbose(
+                        "Calling activity pid:" + AndroidNative.OS.Process.MyPid()
+                        + " tid:" + AndroidNative.OS.Process.MyTid() + "uid:"
+                        + AndroidNative.OS.Process.MyUid());
+
+                    _activity.StartActivityForResult(brokerIntent, 1001);
+                }
+                catch (ActivityNotFoundException e)
+                {
+                    _logger.ErrorPii(e);
```

Comment:
can you add something to the error message, so when we get logs from customers, we know exactly where we are.